src/main/java/org/shanerx/tradeshop/shop/listeners/ShopCreateListener.java

```diff
@@ -119,14 +119,14 @@ public void onSignChange(SignChangeEvent event) {
 
 		shop.setEvent(event);
 
-		ItemStack product = lineCheck(IllegalItemList.TradeItemType.PRODUCT, event.getLine(1)),
-				cost = lineCheck(IllegalItemList.TradeItemType.COST, event.getLine(2));
+		ItemStack product = lineCheck(ShopItemSide.PRODUCT, event.getLine(1)),
+				cost = lineCheck(ShopItemSide.COST, event.getLine(2));
 
-		if (product != null && shop.getProduct().isEmpty())
-			shop.setProduct(product);
+		if (product != null && !shop.hasSide(ShopItemSide.PRODUCT))
+			shop.setSideItems(ShopItemSide.PRODUCT, product);
 
-		if (cost != null && shop.getCost().isEmpty())
-			shop.setCost(cost);
+		if (cost != null && !shop.hasSide(ShopItemSide.COST))
```

Comment:
Should I replace these with an `isSideEmpty1 method so that they are more readable? the method would just return this logic so its completely unnecessary but would be easier to read/check.